### PR TITLE
NAIS env password prompt and secret resolving

### PR DESCRIPTION
**Pull request by TheUnnamedDude**
To avoid having to  save your password as an environment variable I've made it possible to prompt for password. The command will now also fall back to using the $USER environment variable, when this should be enough for most use cases.

The other commit adds support for resolving secrets from fasit.

### CODE REVIEW

File: cli/cmd/environment.go [hunk 1]
```diff
@@ -31,6 +34,22 @@ Or just save it to a file
 		username := os.Getenv("FASIT_USERNAME")
 		password := os.Getenv("FASIT_PASSWORD")
 
+		// Fall back to the ident of the logged in user
+		if username == "" {
+			username = os.Getenv("USER")
```
Comment (Kyrremann) [inline cli/cmd/environment.go]: Would it be more natural to also ask for the username, and set it to `$USER` as default, so you could just press return?

```
Username (default $USER): 
```

Comment (davidsteinsland) [reply cli/cmd/environment.go]: I what circumstances is `$USER` wrong? I would prefer to "silently" default to `$USER` as this removes the need to press Return 99 % of the times.

File: cli/cmd/environment.go [hunk 2]
```diff
@@ -31,6 +34,22 @@ Or just save it to a file
...
+		}
+
+		if password == "" {
+			fmt.Fprintf(os.Stderr, "Enter ident password: ")
```
Comment (davidsteinsland) [inline cli/cmd/environment.go]: I would remove `ident`, as you sometimes can use service users and so on.

Comment (TheUnnamedDude) [reply cli/cmd/environment.go]: I agree, its kinda redundant anyway

Comment (davidsteinsland) [reply cli/cmd/environment.go]: Maybe we could print the value of `username` instead? `Enter password for <username>`?

Comment (Kyrremann) [reply cli/cmd/environment.go]: Printing the username here, removes the need for entering the user earlier. I just want a way to know which user is used.

File: cli/cmd/environment.go (hunk 1 above)
Comment (jhrv) [reply cli/cmd/environment.go]: I misread the code. I agree that this makes sense

File: cli/cmd/environment.go (hunk 2 above)
Comment (TheUnnamedDude) [reply cli/cmd/environment.go]: Makes sense, I'll fix it when I get back home.

File: cli/cmd/environment.go [hunk 3]
```diff
@@ -34,13 +34,13 @@ Or just save it to a file
 		username := os.Getenv("FASIT_USERNAME")
 		password := os.Getenv("FASIT_PASSWORD")
 
-		// Fall back to the ident of the logged in user
+		// Fall back to the username of the logged in user
```
Comment (Kyrremann) [inline cli/cmd/environment.go]: Is this comment necessary?

Comment (TheUnnamedDude) [reply cli/cmd/environment.go]: I think it would be useful for non-unix people reading the code wondering where the USER environment variable comes from

Comment (TheUnnamedDude) [reply cli/cmd/environment.go]: Looking at windows I could do the same thing with the USERNAME environment variable

Comment (davidsteinsland) [reply cli/cmd/environment.go]: my opinion is that we can assume this is basis knowledge, but the behavior still needs to be documented (in README).

When it comes to Windows, maybe we could try adding `if os == windows` before checking the `USERNAME` variant?

**Comment by davidsteinsland** [comment]
Merge?

**Pull request merged**